homeassistant/components/tag/__init__.py

```diff
@@ -0,0 +1,86 @@
+"""The Tag integration."""
+import logging
+import typing
+
+import voluptuous as vol
+
+from homeassistant.const import CONF_NAME
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers import collection
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.storage import Store
+from homeassistant.loader import bind_hass
+import homeassistant.util.dt as dt_util
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+DEVICE_ID = "device_id"
+EVENT_TAG_SCANNED = "tag_scanned"
+LAST_SCANNED = "last_scanned"
+STORAGE_KEY = DOMAIN
+STORAGE_VERSION = 1
+TAG_ID = "tag_id"
+TAGS = "tags"
+
+CREATE_FIELDS = {vol.Required(TAG_ID): cv.string}
+
+UPDATE_FIELDS = {
+    vol.Required(CONF_NAME): vol.All(str, vol.Length(min=1)),
+    vol.Optional("description"): cv.string,
+    vol.Optional(LAST_SCANNED): cv.datetime,
+}
+
+CONFIG_SCHEMA = vol.Schema(
+    {DOMAIN: cv.schema_with_slug_keys(vol.Any(UPDATE_FIELDS, None))},
```

Comment:
No need to enforce slugs, just enforce `str`.

BTW this is for YAML support but there is no YAML collection ?

Reply:
dropped... forgot to remove that when copying from an input_* class